src/routes/api/ticket/due/+server.ts

```diff
@@ -0,0 +1,38 @@
+import { editTicketDueDate, isAdminSession } from '$lib/server/database';
+import { AssertionError } from 'node:assert/strict';
+import type { RequestHandler } from './$types';
+import { StatusCodes } from 'http-status-codes';
+import { error } from '@sveltejs/kit';
+
+// eslint-disable-next-line func-style
+export const PATCH: RequestHandler = async ({ cookies, request }) => {
+    const form = await request.formData();
+
+    const tid = form.get('id');
+    if (tid === null || tid instanceof File) throw error(StatusCodes.BAD_REQUEST);
+
+    const due_date = form.get('due_date');
+    if (due_date === null || due_date instanceof File || new Date(due_date.toString()) < new Date())
+        throw error(StatusCodes.BAD_REQUEST);
+    const date = new Date(due_date.toString());
+
+    const sid = cookies.get('sid');
+    if (!sid) throw error(StatusCodes.UNAUTHORIZED);
+
+    // TODO: session has expired so we must inform the client that they should log in again
+    const head = await isAdminSession(sid);
+    switch (head) {
```

Comment:
`admin` is a more appropriate name here.
```suggestion
    const admin = await isAdminSession(sid);
    switch (admin) {
```